### ISSUE TITLE
Synchronization validation has false positive for multiple subpasses

### ISSUE THREAD

**Issue by HansKristian-Work** [KhronosGroup/Vulkan-ValidationLayers#3493]
**Describe the Issue**
In Granite, the final UI pass can use two subpasses which each render to the backbuffer. I get a validation error on recent validation layers that I haven't seen before:

**Valid Usage ID**

[ERROR]: [Vulkan]: Validation Error: Validation Error: [ SYNC-HAZARD-WRITE_AFTER_WRITE ] Object 0: handle = 0x2602c100000000b8, type = VK_OBJECT_TYPE_RENDER_PASS; | MessageID = 0xfdf9f5e1 | vkCmdEndRenderPass: Hazard WRITE_AFTER_WRITE with last use subpass 1 for attachment 0 final image layout transition (old_layout: VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL, new_layout: VK_IMAGE_LAYOUT_PRESENT_SRC_KHR). Access info (usage: SYNC_IMAGE_LAYOUT_TRANSITION, prior_usage: SYNC_COLOR_ATTACHMENT_OUTPUT_COLOR_ATTACHMENT_WRITE, write_barriers: 0, command: Command_Undefined, seq_no: 4294967295, subcmd: 4294967295, reset_no: 39).

**Environment:**
 - OS: Arch Linux, should be OS agnostic though
 - GPU: RX 6800 or Intel UHD 620 (doesn't seem to matter)
 - SDK or header version if building from repo: Arch Linux package
 ```
 Name            : vulkan-validation-layers
Version         : 1.2.194-1
```
 - Options enabled (synchronization, best practices, etc.): Synchronization validation enabled.

**Additional context**

To reproduce the error, build Granite:
```
git clone git://github.com/Themaister/Granite --recursive
cd Granite
mkdir build
cd build
cmake .. -DCMAKE_BUILD_TYPE=Debug
make gltf-viewer -j16
./viewer/gltf-viewer /path/to/KhronosGroup/glTF-Sample-Models/2.0/Suzanne/glTF/Suzanne.gltf
```
I believe the check is missing something.

![deps](https://user-images.githubusercontent.com/17786731/139559921-9c4ed757-2a51-4fff-8cf4-e66c62e20382.png)

Here's the subpass dependency setup in question, and the attachment:

![attachment](https://user-images.githubusercontent.com/17786731/139559943-ae7d6da7-d0d1-4c67-85c4-d08b40c999ec.png)

Basically, the transition from COLOR -> PRESENT_SRC_KHR is internally synchronized since it's finalLayout handling, and you're not supposed to synchronize that manually. A similar scenario with just one subpass does not trip the error, so I think the sync validation gets confused by the multiple subpasses here.

**Comment by jzulauf-lunarg** [comment]
> the transition from COLOR -> PRESENT_SRC_KHR is internally synchronized since it's finalLayout handling

Not sure why you think that.  Our understanding is that you need appropriate dependencies to VK_SUBPASS_EXTERNAL, with the src*Mask to cover the color and depth attachments you're transitioning and dst*Mask to protect the *next* usage of the transitioned images.

**Comment by jzulauf-lunarg** [comment]
@HansKristian-Work  -- reviewing this.  Yeah, the implicit dependency should be catching this, since you aren't defining one.

**Comment by jzulauf-lunarg** [comment]
Cannot repro.  Hazard no longer detected (by setting breakpoint within validation).  Implicit barrier code has been revised, since report.  Likely fixed by https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/3377

**Comment by jzulauf-lunarg** [comment]
@HansKristian-Work if you're still seeing this, please reopen with updated repro steps.
